File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/EmailAddressColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { EmailAddressRow } from './EmailAddressRow';

export interface EmailAddressColumns {
    BusinessEntityId: Column<EmailAddressRow>;
    EmailAddressId: Column<EmailAddressRow>;
    EmailAddress: Column<EmailAddressRow>;
    IsActive: Column<EmailAddressRow>;
    TenantId: Column<EmailAddressRow>;
}

export class EmailAddressColumns extends ColumnsBase<EmailAddressRow> {
    static readonly columnsKey = 'Company.EmailAddress';
    static readonly Fields = fieldsProxy<EmailAddressColumns>();
}

File: 2025/Company/Company.Web/Modules/Company/EmailAddress/EmailAddressColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.EmailAddress")]
[BasedOnRow(typeof(EmailAddressRow), CheckNames = true)]
public class EmailAddressColumns
{
    public string BusinessEntityPersonType { get; set; }
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int EmailAddressId { get; set; }
    [EditLink]
    public string EmailAddress { get; set; }
    public short IsActive { get; set; }
    public int TenantId { get; set; }
}